src/stcal/ramp_fitting/ramp_fit.py

```diff
@@ -239,15 +247,30 @@ def ramp_fit_data(
         Object containing optional GLS-specific ramp fitting data for the
         exposure
     """
+    # For the LIKELY algorithm, due to the jump detection portion of the code
+    # a minimum of a four group ramp is needed.
+    ngroups = ramp_data.data.shape[1]
+    likely_min_ngroups = 4
     if algorithm.upper() == "GLS":
         image_info, integ_info, gls_opt_info = gls_fit.gls_ramp_fit(
             ramp_data, buffsize, save_opt, readnoise_2d, gain_2d, max_cores
         )
         opt_info = None
+    elif algorithm.upper() == "LIKELY" and ngroups >= likely_min_ngroups:
+        image_info, integ_info, opt_info = likely_fit.likely_ramp_fit(
+            ramp_data, buffsize, save_opt, readnoise_2d, gain_2d, weighting, max_cores
+        )
+        gls_opt_info = None
     else:
         # Default to OLS.
         # Get readnoise array for calculation of variance of noiseless ramps, and
         #   gain array in case optimal weighting is to be done
+        # XXX If the LIKELY is selected, log that the "OLS" algorithm is being use
+        #     and note the minimum number of ngroups needed.
+        if algorithm.upper() == "LIKELY" and ngroups < likely_min_ngroups:
+            msg = f"The 'OLS' algorithm is used since the LIKELY algorithm requires {likely_min_ngroups} or more"
+            msg += f"NGROUPS.  The NGROUPS for this data,{ngroups}, is insufficient."
+            log.warning(msg)
```

Comment:
How do you test to see if a particular message is logged?

Reply:
You can use the caplog fixture.  Here's an example from stcal:
https://github.com/spacetelescope/stcal/blob/c1a8bab38cdfd5221c74df870000183921127e3b/tests/test_alignment.py#L331

Reply:
@kmacdonald-stsci - it looks like you removed the log message here, but it is still needed.  Testing on some data with 3 groups and calling ramp fit directly on a jump output, it just quietly defaults to ols_c without warning, which was confusing.

Reply:
I removed it because it was in the wrong spot (it didn't make sense where it was) and duplicates this:

https://github.com/kmacdonald-stsci/stcal/blob/905ae93dbde265310f44efae396cc229cdf32035/src/stcal/ramp_fitting/likely_fit.py#L61

I think I see how to fix the "quietly" defaulting to `OLS`.

Reply:
Fixed and added test.